src/syntax/read.rkt

```diff
@@ -54,16 +55,17 @@
                               (list-ref args (add1 (index-of args ':precision))))
                         (cons arg-name default-prec))))
 
-  (define ctx-prec (dict-ref prop-dict* ':precision 'binary64))
-  (define type-ctx (map (curryr cons ctx-prec) args))
+  (define default-repr (get-representation default-prec))
+  (define var-reprs (for/list ([(var prec) (in-dict var-precs)])
+                        (cons var (get-representation prec))))
```

Comment:
I think now that we don't use `default-prec` and `var-precs` we can delete those variables entirely and use the `repr` versions.